components/src/core/AppBar/AppBar.tsx

```diff
@@ -0,0 +1,299 @@
+                theme.transitions.create(['all'], {
+                    duration: props.animationDuration || theme.transitions.duration.standard,
+                    easing: theme.transitions.easing.easeInOut,
+                }),
+        },
+        expanded: {},
+        expandedBackground: {},
+    })
+);
+
+export type AppBarProps = MuiAppBarProps & {
+    /**
+     * Length of the collapse / expand animation (in ms)
+     * Default: 300
+     */
+    animationDuration?: number;
+
+    /**
+     * Image to use as the background for the header
+     * Default: none
+     */
+    backgroundImage?: string;
+
+    /**
+     * Opacity to use for the header background image
+     * Default: 0.3
+     */
+    // backgroundImageOpacity?: number;
+
+    /**
+     * Custom classes to add to app bar elements
+     * Default: none
+     */
+    classes?: Partial<AppBarClasses>;
+
+    /**
+     * Height of the App Bar when fully collapsed
+     * Default: 64 desktop, 56 mobile
+     */
+    collapsedHeight?: number;
+
+    /**
+     * Height of the App Bar when fully expanded
+     * Default: 200
+     */
+    expandedHeight?: number;
+
+    /**
+     * Current mode of the app bar:
+     * - 'expanded' locks the app bar at the expandedHeight,
+     * - 'collapsed' locks it at the collapsedHeight,
+     * - 'dynamic' resizes the toolbar based on the window scroll position.
+     * Default: dynamic
+     */
+    mode?: 'expanded' | 'collapsed' | 'dynamic';
+
+    /**
+     * How far to scroll before collapsing the app bar
+     * Default: 50
+     */
+    scrollThreshold?: number;
+
+    /**
+     * A ref to the scrollable container that controls the app bar height
+     * Default: window
+     * TODO: NOT IMPLEMENTED YET
+     */
+    // scrollRef?: MutableRefObject<any>;
+
+    /**
+     * NOT EXPOSED YET
+     * Callback when the dynamic toolbar has reached it's expandedHeight
+     * Default: none
+     */
+    // onExpandedHeightReached?: () => void;
+
+    /**
+     * NOT EXPOSED YET
+     * Callback when the dynamic toolbar has reached it's collapsedHeight
+     * Default: none
+     */
+    // onCollapsedHeightReached?: () => void;
+};
+
+const AppBarRender: React.ForwardRefRenderFunction<unknown, AppBarProps> = (props: AppBarProps, ref: any) => {
+    const {
+        style = {},
+        mode = 'dynamic',
+        animationDuration: durationProp,
+        expandedHeight = 200,
+        backgroundImage,
+        classes = {},
+        collapsedHeight: collapsedHeightProp,
+        // onExpandedHeightReached,
+        // onCollapsedHeightReached,
+        scrollThreshold = 100,
+        ...muiAppBarProps
+    } = props;
+    const theme = useTheme();
+    const defaultClasses = useStyles(props);
+    const animationDuration = durationProp || theme.transitions.duration.standard;
+    const isMobile = useMediaQuery(theme.breakpoints.down('xs'));
+    const defaultAppBarHeight = isMobile ? theme.spacing(7) : theme.spacing(8);
```

Comment:
If you want to go fancy about it, `theme.mixins.toolbar` is what you shall be looking into. https://material-ui.com/customization/default-theme/

Reply:
`theme.mixins.toolbar['media (min-width:0px) and (orientation: landscape)'].minHeight` doesn't feel quite as clean...I presume it would also not work if somebody changes the default breakpoints in their theme. I'm actually going to tweak this to use REMS so it will exactly match what we're doing in the DrawerHeader